project4/name_server/tasks/nameserver_task.cpp

```diff
@@ -4,18 +4,16 @@
 
 namespace nameserver::tasks {
 
-/// TODO Instantiate name server
-NameserverTask::NameserverTask(std::shared_ptr<nameserver::Nameserver> nameserver)
+NameserverTask::NameserverTask(
+    std::shared_ptr<nameserver::Nameserver> nameserver)
     : nameserver_(nameserver) {}
 
-/// TODO Set up socket listening
-thread_pool::Task::Status NameserverTask::SetUp() {
-  return Task::SetUp();
-}
-
-/// TODO Listen to all requests. Pass them accordingly to HandleResponse
-/// If there is a command in cmd_queue_, parse and execute.
 thread_pool::Task::Status NameserverTask::RunAtomic() {
+  message_passing::Endpoint endpoint;
```

Comment:
We should really combine these into one class. Have you tried simply substituting a `NameserverTask` where you would otherwise use a `BootstrapTask`? I bet this would work, since `Bootstrap` is a subclass of `Nameserver`.

Reply:
You're right, combining these would be best. I haven't yet tried it that way, but it is likely doable.